src/io/exporters/FBXInfoExtractor.js

```diff
@@ -0,0 +1,324 @@
+        colorEnd.fromArray(color.array, colorIdx);
+        twoCCylinder.setColors(colorStart, colorEnd);
+        geo = twoCCylinder;
+      } else {
+        // has one color per cylinder
+        colorStart.fromArray(color.array, colorIdx);
+        oneCCylinder.setColors(colorStart);
+        geo = oneCCylinder;
+      }
+      // add instance to the model
+      this._getCylinderInstanceMatrix(mesh.geometry, instanceIndex, instMatrix);
+      objMatrix.multiplyMatrices(matrix, instMatrix);
+      model.addInstance(objMatrix, geo);
+    }
+    const material = this._collectMaterialInfo(mesh);
+    this._addToPool(model, material);
+  }
+
+  /**
+   * Adding model to pool of models or extend existing ones
+   * @param {object} model - model to add
+   * @param {object} material - material to add
+   */
+  _addToPool(model, material) {
+    const materialIdx = this._checkExistingMaterial(material);
+    if (materialIdx === this._models.length) { // new model-material pair
+      model.reworkIndices();
+      this._models.push(model);
+      this._materials.push(material);
+    } else { // add model to existing model-material pair
+      const oldModel = this._models[materialIdx];
+      oldModel.concatenate(model);
+    }
+  }
+
+  /**
+   * Checking if given material already was registered in materials pool (no need to create new one)
+   * @param {object} material - given material
+   * @returns {number} number of model-material pair
+   */
+  _checkExistingMaterial(material) {
+    for (let i = 0; i < this._materials.length; ++i) {
+      if (_.isEqual(material, this._materials[i])) {
+        return i;
+      }
+    }
+    return this._models.length;
+  }
+
+  _gatherCylindersColoringInfo(geo) {
+    const instCount = geo.maxInstancedCount;
+    const color1 = geo.attributes.color.array;
+    const color2 = geo.attributes.color2.array;
+    const stride = geo.attributes.color.itemSize;
+    const is2Colored = new Array(instCount);
+    // analyze color instance attributes
+    let needToSplit = 0;
+    let colIdx = 0;
+    for (let i = 0; i < instCount; i++, colIdx += stride) {
+      const differs = (Math.abs(color1[colIdx] - color2[colIdx]) > 0.0000001) // compare ending colors
+        || (Math.abs(color1[colIdx + 1] - color2[colIdx + 1]) > 0.0000001)
+        || (Math.abs(color1[colIdx + 2] - color2[colIdx + 2]) > 0.0000001);
+      is2Colored[i] = differs;
+      needToSplit += differs; // count number of 2-colored cylinders
+    }
+    // calc number of vertices to add into 2-colored
+    const geoParams = geo.getGeoParams();
+    const addPerCylinder = geoParams.radialSegments;
+    return { is2Colored, needToSplit, addPerCylinder };
+  }
+
+  /**
+   * Collect instanced models and materials.
+   * @param {object} mesh - given mesh with instanced something (spheres or cylinders)
+   */
+  _collectInstancedGeoInfo(mesh) {
+    if (mesh.geometry.constructor.name.includes('Spheres')) {
```

Comment:
I'm afraid this should not work because after uglifying the function names change. Do we have another way to perform the necessary test?

Reply:
Thanks, you are absolutely right. I've fixed the issue.